consensus/istanbul/backend/backend.go

```diff
@@ -554,6 +554,13 @@ func (sb *Backend) Commit(proposal istanbul.Proposal, aggregatedSeal types.Istan
 		}
 	}
 	sb.onNewConsensusBlock(block, result.Receipts, result.Logs, result.State)
+
+	nextBlockNum := new(big.Int).Add(block.Number(), big.NewInt(1))
+	if sb.chain.Config().IsL2Migration(nextBlockNum) {
+		sb.logger.Info("The next block is the L2 migration block, stopping announce protocol and closing istanbul backend", "currentBlock", block.NumberU64(), "hash", block.Hash(), "nextBlock", nextBlockNum)
+		sb.StopAnnouncing()
+		sb.Close()
```

Comment:
I don't think this is safe to call from here since Close is also called from other threads, it might be enough to add a lock inside Close so that only one thread can be executing in there at any time. Needs a bit more thought though.

Reply:
Do we even need to call Close though?

Reply:
It looks like calling Close() may be a bit overkill actually. StopAnnouncing() stops the announce protocol and it looks like Close() basically prevents it from being easily restarted with StartAnnouncing(). I removed it and the e2e test still passed so I'll remove it from the PR